scrapi/migrations.py

```diff
@@ -19,97 +14,87 @@
 @tasks.task_autoretry(default_retry_delay=30, max_retries=5)
 def rename(docs, target=None, **kwargs):
     assert target, "To run this migration you need a target."
+
     for doc in docs:
-        raw = RawDocument({
-            'doc': doc.doc,
-            'docID': doc.docID,
-            'source': target,
-            'filetype': doc.filetype,
-            'timestamps': doc.timestamps,
-            'versions': doc.versions
-        })
+        raw_doc = doc.raw
+        new_doc = copy.copy(raw_doc.attributes)
+        new_doc['source'] = target
 
-        assert doc.source != target, "Can't rename {} to {}, names are the same.".format(doc.source, target)
+        raw = RawDocument(new_doc, validate=False)
+
+        assert raw_doc.attributes['source'] != target, "Can't rename {} to {}, names are the same.".format(raw_doc['source'], target)
 
         if not kwargs.get('dry'):
             tasks.process_raw(raw)
             tasks.process_normalized(tasks.normalize(raw, raw['source']), raw)
-            logger.info('Processed document from {} with id {}'.format(doc.source, raw['docID']))
+            logger.info('Processed document from {} with id {}'.format(raw_doc.attributes['source'], raw['docID']))
+
+            es_processor = get_processor('elasticsearch')
+            es_processor.manager.es.delete(index=settings.ELASTIC_INDEX, doc_type=raw_doc.attributes['source'], id=raw['docID'], ignore=[404])
+            es_processor.manager.es.delete(index='share_v1', doc_type=raw_doc.attributes['source'], id=raw['docID'], ignore=[404])
+
+        logger.info('Renamed document from {} to {} with id {}'.format(raw_doc.attributes['source'], target, raw['docID']))
+
+
+@tasks.task_autoretry(default_retry_delay=30, max_retries=5)
+def cross_db(docs, target_db=None, index=None, **kwargs):
+    """
+    Migration to go between
+        cassandra > postgres
+        postgres > cassandra
+        cassandra > elasticsearch
+        postgres > elasticsearch
+
+    source db can be passed in to the migrate task, and will default to the CANONICAL_PROCESSOR specified in settings
+    target_db will be specified when the task is called
+    """
+    assert target_db, 'Please specify a target db for the migration -- either postgres or elasticsearch'
+    assert target_db in ['postgres', 'cassandra', 'elasticsearch'], 'Invalid target database - please specify either postgres, cassandra or elasticsearch'
+    for doc in docs:
+        if not doc.raw.attributes['doc']:
```

Comment:
I think `doc.raw['doc']` should work. We should probably avoid accessing the internals of `Raw/NormalizedDocument`s where available (structure could easily change in the future...)